bin/node/runtime/src/impls.rs

```diff
@@ -247,77 +243,112 @@ mod tests {
 	fn stateless_weight_mul() {
 		run_with_system_weight(target() / 4, || {
 			// Light block. Fee is reduced a little.
```

Comment:
It is great that you have made these much more accureate. But to uphold to this (main) criteria, would be good to see an assertion like:

```rust
assert!(next_fee_multiplier < 1) // meaning that it is **reduced** since `target() / 4` is filled. 
```

Reply:
(It it kinda hard to read this from the two different contractors that you use for Fixed128)

Reply:
I actually wouldn't mind if you'd do the accurate calculation only for one and for the rest only assert on the _trend_.

Reply:
I think you mean in this context the fee adjustment is positive or negative right?

This is how I have re-written the test:

```rust
	#[test]
	fn stateless_weight_mul() {
		// This test will show that heavy blocks have a weight multiplier greater than 0
		// and light blocks will have a weight multiplier less than 0.
		run_with_system_weight(target() / 4, || {
			// Target = 25%, Weight is Target / 4 = 6.25%, Diff = -18.75%, V = 4/10^5
			// So first term is -75/10^7
			let first_term = Fixed128::from_rational(-75, NonZeroI128::new(10_000_000).unwrap());
			// Diff^2 = .03515625, V^2 = 8/10^10... So second_term = 28125/10^15
			let second_term = Fixed128::from_parts(28_125_000);
			assert_eq!(
				TargetedFeeAdjustment::<TargetBlockFullness>::convert(Fixed128::default()),
				first_term + second_term,
			);

			// Light block. Fee is reduced a little.
			let target_fee_adjustment = TargetedFeeAdjustment::<TargetBlockFullness>::convert(Fixed128::default());
			assert!(target_fee_adjustment < Fixed128::zero())
		});
		run_with_system_weight(target() / 2, || {
			// a bit more. Fee is decreased less, meaning that the fee increases as the block grows.
			let target_fee_adjustment = TargetedFeeAdjustment::<TargetBlockFullness>::convert(Fixed128::default());
			assert!(target_fee_adjustment < Fixed128::zero());
			// -74/10^7 is greater than the target/4 fee adjustment, and this should be even bigger.
			assert!(target_fee_adjustment > Fixed128::from_rational(-74, NonZeroI128::new(10_000_000).unwrap()))

		});
		run_with_system_weight(target(), || {
			// ideal. Original fee. No changes.
			let target_fee_adjustment = TargetedFeeAdjustment::<TargetBlockFullness>::convert(Fixed128::default());
			assert_eq!(target_fee_adjustment, Fixed128::zero())
		});
		run_with_system_weight(target() * 2, || {
			// More than ideal. Fee is increased.
			let target_fee_adjustment = TargetedFeeAdjustment::<TargetBlockFullness>::convert(Fixed128::default());
			assert!(target_fee_adjustment > Fixed128::zero())
		});
	}